Model/RatepayInvoicePaymentMethod.php

```diff
@@ -278,9 +281,22 @@ public function addMandatoryTransactionData(&$oTransaction, $oOrder)
      */
     public function isPaymentPossible()
     {
-        // no need to handle if basket amount is within range, this is checked by oxid
-        // TODO: add additional checks as soon as values are available
-        return !SessionHelper::isDateOfBirthSet() || SessionHelper::isUserOlderThan(18);
+        $oSession = Registry::getSession();
+        $oBasket = $oSession->getBasket();
+        $sBillingCountryId = $oSession->getUser()->oxuser__oxcountryid->value;
+        $sShippingCountryId = $sBillingCountryId;
+
+        if ($oSession->getVariable('deladrid')) {
```

Comment:
Can we make a constant and use it thro the plugin. Also please encapsulate the two ifs into separate private methods. In my opinion its also better to first check if the billing and shipping needs to be same, if not you just load the shipping address. This logic belongs in a private function like you did with _areArticlesAllowed and then also be called in the method _shouldBeDisplayed

Reply:
I split the logic up into two helper methods now placed in `SessionHelper`. As for the different shipping & billing country, this is now handled in `_areAddressesAllowed`.